app/src/main/java/com/example/testgit/MainActivity.java

```diff
@@ -39,31 +54,52 @@ public void createMap() {
 
             public boolean onTouch(View arg0, MotionEvent event) {
 
-                float curX, curY;
-                int mx = 0;
-                int my = 0;
+                int curX, curY;
+                //Log.i();
+                //int maxX = 0;
+                //int minY = 0;
+
                 //These are original x and y coordinates
 
                 switch (event.getAction()) {
 
                     case MotionEvent.ACTION_DOWN:
-                        mx = (int) event.getX();
-                        my = (int) event.getY();
+                        downX = (int) event.getX();
+                        downY = (int) event.getY();
+                        Log.d(LOG_TAG, String.format("ad: downY == %x", downY));
+                        Log.d(LOG_TAG, String.format("ad: downX == %x", downX));
+
+
                         //x and y cords initialized
+
+
+                        //switcherView.setMaxHeight(100);
+                        //switcherView.setMaxWidth(100);
+
+                        //Set max cords
+
                         break;
                     case MotionEvent.ACTION_MOVE:
-                        curX = event.getX();
-                        curY = event.getY();
-                        switcherView.scrollBy((int) (mx - curX), (int) (my - curY));
+                        curX = (int) event.getRawX();
+                        curY = (int) event.getRawY();
+
+                        switcherView.scrollBy( (downX - curX), (downY - curY));
                         //scrollBy -
```

Comment:
We do not need to cast downX and downY as ints because curX and curY are already ints